### ISSUE TITLE
[BUG] Crash after 60s

### ISSUE THREAD

**Issue by kamaradclimber** [joshuar/go-hass-agent#5]
**Describe the bug**

After starting go-hass-agent, it crashes after ~60s

**To Reproduce**

Launch `go-hass-agent` binary.

**Expected behavior**

The process should remain up until interrupted.

**Screenshots**

**Logs**
A log will be very helpful to look into this bug report. To get the log:

```
6:08PM DBG Debug logging enabled.
6:08PM INF Starting agent.
6:08PM DBG internal/agent/config.go:74 > Setting language to [en-US].
6:08PM DBG internal/agent/config.go:53 > Using URL https://ha.familleseux.net:443/api/webhook/b1dbfd47be7bcf0852b096a3ed90535667f11e814434f3397a7a95a825b8d860 for Home Assistant access
6:08PM DBG internal/agent/config.go:53 > Using URL https://ha.familleseux.net:443/api/webhook/b1dbfd47be7bcf0852b096a3ed90535667f11e814434f3397a7a95a825b8d860 for Home Assistant access
6:08PM DBG internal/device/device.go:84 > Registering Battery sensors.
6:08PM DBG internal/device/device.go:84 > Registering Apps sensors.
6:08PM DBG internal/device/device.go:84 > Registering Network sensors.
6:08PM DBG internal/agent/agent.go:158 > Setting up sensors for Battery.
6:08PM DBG internal/agent/agent.go:158 > Setting up sensors for Apps.
6:08PM DBG internal/agent/agent.go:158 > Setting up sensors for Network.
6:08PM DBG internal/hass/websocket.go:98 > Using wss://ha.familleseux.net:443/api/websocket for websocket connection.
6:08PM WRN Unsupported desktop/window environment.
6:08PM ERR Unable to execute org.freedesktop.NetworkManager.GetDevices on org.freedesktop.NetworkManager (args: ) error="The name org.freedesktop.NetworkManager was not provided by any .service files"
6:08PM DBG internal/device/network_linux.go:352 > Could not list devices from network manager
6:08PM DBG internal/device/apps_linux.go:126 > Unsupported or unknown portal
2023/04/12 18:08:46 systray error: failed to register our icon with the notifier watcher (maybe no tray is running?): The name org.kde.StatusNotifierWatcher was not provided by any .service files
6:08PM DBG internal/agent/config.go:53 > Using URL https://ha.familleseux.net:443/api/webhook/b1dbfd47be7bcf0852b096a3ed90535667f11e814434f3397a7a95a825b8d860 for Home Assistant access
6:08PM DBG internal/agent/notifications.go:45 > Requesting authorisation for websocket.
6:08PM DBG internal/agent/notifications.go:54 > Registering app for push notifications.
2023/04/12 18:09:46 Message recipient disconnected from message bus without replying
```

it is likely relevant to note in journalctl at the same time

```
Apr 12 18:09:46 gagarine geoclue[1609514]: Service not used for 60 seconds. Shutting down..
Apr 12 18:09:46 gagarine systemd[1]: geoclue.service: Deactivated successfully.
```

**Desktop (please complete the following information):**
 - OS: Archlinux
 - Version 0.0.2 + #4

**Comment by kamaradclimber** [comment]
I assume the issue is due to my geoclue setup not working properly. I would expect the agent not to crash in that situation though.

**Comment by joshuar** [comment]
Hey @kamaradclimber thanks for the report. It seemed a bit tricky to handle this situation as the library I'm using to get the location via geoclue is a bit wonky with reporting errors it seems. I've got a workaround in https://github.com/joshuar/go-hass-agent/commit/72d6541cfcd264222024fe348dd278860595d957 (release 0.0.3) that seemed to work in an Arch container I tested on with distrobox. I unfortunately don't have access to a full Arch install so I'm hoping this is a fix for you as well.
